packages/http/tools/e2e-tests/src/warrior/middleware/middlewares/UnsuccessfulMiddleware.ts

```diff
@@ -0,0 +1,28 @@
+import { HttpStatusCode, Middleware } from '@inversifyjs/http-core';
+import { injectable } from 'inversify';
+
+@injectable()
+export class UnsuccessfulMiddleware
+  implements Middleware<unknown, unknown, unknown, unknown>
+{
+  public async execute(
+    _request: unknown,
+    response: {
+      send?: () => unknown;
+      text?: () => unknown;
+      status: (code: number) => void;
+    },
```

Comment:
This implementation might cause trouble with new adapters. I think we need a class per integration or a way to detect which adapter we are using and use the right specific adapter types